test/extended/prometheus/prometheus_builds.go

```diff
@@ -133,6 +134,10 @@ func runQueries(metricTests map[string][]metricTest, oc *exutil.CLI, ns, execPod
 
 			// for each test case, register that one of the returned metrics has the desired labels and value
 			for j, tc := range tcs {
+				if tcs[j].nodata && len(metrics) == 0 {
+					tcs[j].success = true
+					break
+				}
```

Comment:
can you quickly elaborate why this is needed?

Reply:
This is a check for when no metrics were reported and that was expected by us.